server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java

```diff
@@ -238,19 +249,35 @@ private void createBlockMessageSpan(BlockLogMessage blockLogMessage, Span buildS
             Span childSpan = this.otelHelper.createTransientSpan(blockMessageStepName, parentSpan, blockLogMessage.getTimestamp().getTime());
             blockMessageSpanMap.put(blockMessageStepName, childSpan);
             Loggers.SERVER.debug("OTEL_PLUGIN: Build step span added for " + blockMessageStepName);
-            setBlockMessageSpanAttributes(blockLogMessage, childSpan);
+            setBlockMessageSpanAttributes(blockLogMessage, childSpan, build);
             childSpan.end(blockMessageFinishDate.getTime(),TimeUnit.MILLISECONDS);
         }
     }
 
-    private void setBlockMessageSpanAttributes(BlockLogMessage blockLogMessage, Span span) {
+    private void setBlockMessageSpanAttributes(BlockLogMessage blockLogMessage, Span span, SRunningBuild build) {
         if (blockLogMessage.getBlockDescription() != null) {
             // Only the Build Step Types "teamcity-build-step-type" has blockDescriptions
             this.otelHelper.addAttributeToSpan(span, PluginConstants.ATTRIBUTE_NAME, blockLogMessage.getText() + ": " + blockLogMessage.getBlockDescription());
         } else {
             this.otelHelper.addAttributeToSpan(span, PluginConstants.ATTRIBUTE_NAME, blockLogMessage.getText());
         }
         this.otelHelper.addAttributeToSpan(span, PluginConstants.ATTRIBUTE_SERVICE_NAME, blockLogMessage.getBlockType());
+        if (blockLogMessage.getBlockType().contains("checkout")) {
```

Comment:
`contains` feels a bit too loose here - can we check for an exact match?

Reply:
Yep I can make this an equals